docs/self-hosted/config-options.md

```diff
@@ -446,6 +446,20 @@ RATE_LIMITER_REDIS_PORT=6379
 RATE_LIMITER_REDIS_DB=0
 ```
 
+### Pressure-based rate limiter
+
+This rate-limiter prevents the API from accepting new requests while the server is experiencing high load. This continuously monitors the current event loop and memory usage, and error out requests with a 503 early when the system is overloaded.
+
+| Variable                                      | Description                                                         | Default Value |
+|-----------------------------------------------|---------------------------------------------------------------------|---------------|
+| `PRESSURE_LIMITER_ENABLED`                    | Whether or not to enable pressure-based rate limiting on the API.   | `true`        |
+| `PRESSURE_LIMITER_SAMPLE_INTERVAL`            | The time window for measuring pressure in ms.                       | `250`         |
+| `PRESSURE_LIMITER_MAX_EVENT_LOOP_UTILIZATION` | The maximum allowed utilization where `1` is 100% loop utilization. | `0.99`        |
+| `PRESSURE_LIMITER_MAX_EVENT_LOOP_DELAY`       | The maximum amount of time the current loop can be delayed in ms.   | `500`         |
+| `PRESSURE_LIMITER_MAX_MEMORY_RSS`             | The maximum allowed memory Resident Set Size (RSS) in bytes.        | `false`       |
+| `PRESSURE_LIMITER_MAX_MEMORY_HEAP_USED`       | The maximum allowed heap usage in bytes.                            | `false`       |
+| `PRESSURE_LIMITER_RETRY_AFTER`                | Sets the `Retry-After` header when the rate limiter is triggered.   | `false`       |
```

Comment:
What's the reason for having `Retry-After` header disabled by default? Security concerns?

Reply:
Tim's concerns 🙂 

https://github.com/directus/directus/pull/17873#discussion_r1146295732

The TL;DR is: the retry-header isn't actually connected to anything. It's an arbitrary number that just allows the server to breath for a second before the client retries, assuming that the client checks for that header.

Reply:
All clear, thanks!